tests/python/pants_test/reporting/test_reporting_integration.py

```diff
@@ -237,6 +237,27 @@ def test_zipkin_reporter_with_zero_sample_rate(self):
       num_of_traces = len(ZipkinHandler.traces)
       self.assertEqual(num_of_traces, 0)
 
+  def test_zipkin_reporter_for_v2_engine(self):
+    ZipkinHandler = zipkin_handler()
+    with http_server(ZipkinHandler) as port:
+      endpoint = "http://localhost:{}".format(port)
+      command = [
+        '--reporting-zipkin-endpoint={}'.format(endpoint),
+        '--reporting-zipkin-trace-v2',
+        'cloc',
+        'src/python/pants:version'
+      ]
+
+      pants_run = self.run_pants(command)
+      self.assert_success(pants_run)
+
+      num_of_traces = len(ZipkinHandler.traces)
+      self.assertEqual(num_of_traces, 1)
```

Comment:
There's a method `assert_single_element()` I added a while ago for cases like this: https://github.com/pantsbuild/pants/blob/6744c9d12ad6d9f73bac77e750a9319557d0ae94/src/python/pants/util/collections.py#L54-L55
Here, you could write:
```python
from pants.util.collections import assert_single_element
# ...
   trace = assert_single_element(ZipkinHandler.traces)
```
which would raise an exception if there was not exactly 1 trace.

You do not need to use it at all, I have just found it useful to make it more clear what is being tested.